src/sources/kubernetes/mod.rs

```diff
@@ -65,8 +75,16 @@ impl SourceConfig for KubernetesConfig {
 
         let now = TimeFilter::new();
 
-        let (file_recv, file_source) =
-            file_source_builder::FileSourceBuilder::new(self).build(name, globals, shutdown)?;
+        let vector_pod_uid = env::var(VECTOR_POD_UID_ENV).map_err(|error| BuildError::PodUid {
```

Comment:
I know I asked this before but is this env var always available? If it is lets make sure we document this very clearly, if not we need to find a way to work around this and that may be just ignoring this and letting vector logs come through.

Reply:
Also curious.

Reply:
It is present if `.yaml` is configured with 
```yaml
       env:
        - name: VECTOR_POD_UID
          valueFrom:
            fieldRef:
              fieldPath: metadata.uid
```
, and if it isn't configured like that, I wouldn't consider it a proper configuration. In the same way it isn't a proper configuration to run vector without mapping the folders necessary for it's operation.

Passing the logs would make it easy for us brick the node, as it happened with Github's nodes, if we accidentally, or not, add a info log somewhere that depends on the number of logs passing. And even worse if the size of the info depends on the size of the log.

For other ways, we filtered out logs for containers which had names starting with vector, which could be suprising for users that name their containers that way, but we could name vector container in some unique way, but then we are back at the beginning where we need the `.yaml` to be configured in a proper way.

There is also another way, that just poped into my head, we could with intention log a message with unique string of characters and once we detect that string as a message in the log we would know which UID is ours and could filter them out, although file source would still unecessary be picking them up from the log file. 

What do you think?

Reply:
I am not really happy with this required env var personally. I think its a pretty bad UX hazard.

Do we know what other solutions do here? What does fluentd do?

Reply:
For example this daemonset seems quite simple compared to ours https://github.com/fluent/fluentd-kubernetes-daemonset/blob/master/fluentd-daemonset-elasticsearch.yaml

Reply:
Then they are doing something in the code, I'll look into it. My hunch is that they are using label `k8s-app: fluentd-logging` to filter out their own logs.

Reply:
That seems like a good solution possibly, nuch better than an env var imo.

Reply:
In that case we can only filter out our own logs if kubernetes_metadata_transform is used. Since only then do we know the labels of the pods.

Reply:
Is that what fluentd requires?

Reply:
Yes, they are using a combination of tail plugin, which seams to be generic, and metadata transform. [link](https://docs.fluentbit.io/manual/installation/kubernetes)

But there is a more probable way they are doing it, all their `.yaml` have fluent running in `kube-system` namespace, which needs also to be excluded. So they are probably excluding on the basis of the namespace. That doesn't require metadata_transform, but does require Kubernetes v1.14 or greater.

Reply:
The solution with unique string of characters could also be used in `docker` source.

Reply:
That seems like a totally feasible trade off! Can we explore that?

Reply:
Definitely. I'll open a separate issue for it.

Reply:
FYI, what I mean feasible is having vector run in the kube-system namespace. We could also provide a config to ignore certain namespaces. So a user could setup their own vector namespace and match that in their config and vector will ignore those events. What do you think?

Reply:
Yes! I am very much in favor of anything that improves reliability and confidence for v1 of our Kubernetes integration. I do not want to compromise that supporting old versions right now. If requiring >= 1.14 achieves that then let's do it.

> AWS EKS still supports v1.13.

That's ok. EKS [also offers 1.14 and 1.15](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html).

Reply:
I'll add my 2 cents, we've been using in production for quite some time fluentd and there has been an attempt to use fluent-bit, and now we're replacing fluent-bit with vector. In all our configurations we've been deploying the entire log-related stack in its own namespace. Mostly because with an appropriated Role and PSP it can be easily restricted.
Even with the current (0.8.2) kubernetes source it works perfectly fine (because you can just whitelist any namespace except vector's).
Just saying that it can be a suggested pattern of deployment and in this case the solution can be less invasive. Otherwise filtering using Pod's UID can be opted-in.

Reply:
@Alexx-G we really appreciate your feedback! So if I understand correctly for you all it would make sense to have some way in the vector config to set the namespace you are deploying your log stack in. From this we can ignore all logs from that namespace? This to me seems like the ideal solution.

Reply:
Using a separate namespace is a particular deployment pattern, and not everyone can use that. So we can't solely rely on that - it would be a very significant limiting factor.

As an alternative to env vars, we could use downward API mounts: https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/

We can also communicate with the cluster API to determine the current container info: https://kubernetes.io/docs/tasks/administer-cluster/access-cluster-api/#accessing-the-api-from-within-a-pod

Also, a `hostname` in the container environment is by default set to the pod name.

That said, I like the explicit configuration as we do it now. This, being explicit, is much easier to understand and maintain for infra teams. I.e. when they operate on a configuration for vector - it becomes very simple to reason which logs are ignored and which are picked up.
Also, since the industry standard in k8s-compatible apps is to ship the deployment configuration as part of the release, I don't see this configuration detail as particularly problematic. We should provide a helm chart for our stuff anyway, as well as applicable daemonset/sidecar deployment files. This would be a proper equivalent of our "good defaults" paradigm in the k8s sense.